### PR TITLE
Use strict wire DN and registry-first OID parse

### LINKED ISSUE #1863: Subject DNs using platform RDN codes (EMAIL) are rejected on re-parse and by strict RFC 4514 connectors
### Description

The platform renders subject DNs with its OID-registry codes (e.g. EMAIL for 1.2.840.113549.1.9.1 per SystemOid), but two surfaces reject exactly that form:

1. **Parse side (core):** `PlatformX500NameStyle` extends `BCStrictStyle` and does not consult the OID registry when parsing, so a registration subject DN like `EMAIL=mail@mail.com, CN=reg` — the same form the platform itself displays — fails with HTTP 422 "Unknown object id - EMAIL - passed to distinguished name" (`CertificateUtil.applyRegistrationSubject`, `X509RequestContentParser.parseSubjectDn`). Runtime-registered custom OID codes are likewise unparseable, and `SN` (rendered for surname 2.5.4.4) re-parses as serialNumber 2.5.4.5, silently corrupting round-trips.

2. **Wire side (v3 register):** `RegisterWireBuilder` renders the flat `subjectDn` for connectors with the platform display style, emitting non-RFC 4514 keywords (`EMAIL=`) with no value escaping. Strict connector parsers reject it — otpki returns HTTP 400 'Invalid RFC 4514 Subject DN: unknown attribute type "EMAIL"'. Comma-containing values (`O=Org, s.r.o.`) also produce an ambiguous unescaped wire string.

### Steps to Reproduce

1. Register a certificate (no-CSR path) with subject DN `EMAIL=mail@mail.com, CN=reg` → HTTP 422
2. With parsing fixed, register via an RA profile backed by a strict RFC 4514 connector (otpki) → HTTP 400 on the register wire

### Expected Behavior

DN strings the platform renders are accepted back by the platform, and the connector wire carries a strict RFC 4514/2253 subject DN (registered keywords or dotted-decimal OIDs with hexstring values, escaped).

### Actual Behavior

HTTP 422 "Invalid subject DN ... Unknown object id - EMAIL" on core parse; HTTP 400 "Invalid RFC 4514 Subject DN: unknown attribute type \"EMAIL\"" from connector otpki.

### Severity

Major

### Module

Certificates

### PR DESCRIPTION
**Pull request by klaraf755**
**Review-only PR — do not merge.**

The "strict wire DN and registry-first OID parse" change was unintentionally swept into `main` inside the #1839 squash-merge (the branch had merged a local main carrying it), so it never got its own review. This PR reconstructs it for review:

- **Base** (`wire-dn-review-base`): current `main` with the change reverted
- **Head**: base + the original change (`047e46d5a`) + its volatile→AtomicReference follow-up, so the reviewed `OidHandler` state is identical to what `main` has today

Review and comment as usual; when done, close this PR (the content is already in `main`) and delete both branches.

### CODE REVIEW

File: src/main/java/com/otilm/core/util/PlatformX500NameStyle.java [hunk 1]
```diff
@@ -35,6 +35,21 @@ public PlatformX500NameStyle(boolean normalizedStyle) {
...
+     * built-in keyword table so that every code this style renders (default and alternative codes)
+     * parses back. The registry lookup is live — unlike the {@link #oidToCodeMap} snapshot —
+     * because the shared {@link #DEFAULT}/{@link #NORMALIZED} instances outlive runtime
+     * registrations of custom OID entries. Codes match case-insensitively (RFC 4514 keywords are
+     * case-insensitive); unknown codes fall through to BouncyCastle, which also handles the
+     * dotted-decimal form.
+     */
+    @Override
+    public ASN1ObjectIdentifier attrNameToOID(String attrName) {
+        String oid = OidHandler.getOidForRdnCode(attrName);
```
Comment (lubomirw) [inline src/main/java/com/otilm/core/util/PlatformX500NameStyle.java:49]: **[Medium]** | Correctness

Parse path `attrNameToOID` → `OidHandler.getOidForRdnCode` matches RDN codes **case-insensitively**, but the wire render path `X509RequestContentRenderer.resolveOid` (~line 210) uses `getCodeToOidMap().containsKey(type)` — a **case-sensitive** HashMap. A lowercase structured code (e.g. `cn`) parses from a DN string yet throws `Unknown RDN type/code` on the new `renderWireSubjectDn` path. Normal string-parse flow canonicalizes via `rdnTypeCode`, so the trigger is structured `RdnEntry` input with non-canonical case.

_Suggested:_ route render-side resolution through `OidHandler.getOidForRdnCode` (or back `getCodeToOidMap` with a CASE_INSENSITIVE map) so both DN paths share one matching rule.

<sub>flagged by Copilot CLI + guided-pr-review</sub>

File: src/main/java/com/otilm/core/util/PlatformX500NameStyle.java [hunk 2]
```diff
@@ -35,6 +35,21 @@ public PlatformX500NameStyle(boolean normalizedStyle) {
...
+        return oid != null ? new ASN1ObjectIdentifier(oid) : super.attrNameToOID(attrName);
```
Comment (lubomirw) [inline src/main/java/com/otilm/core/util/PlatformX500NameStyle.java:50]: **[Medium]** | Correctness

Registry-first resolution makes `SN` resolve to surname (`2.5.4.4`) instead of BouncyCastle's serialNumber (`2.5.4.5`). This is deliberate and tested (`parsesSnAsSurnamePerRegistry`) for platform round-trip fidelity — flagging for confirmation: DN strings from standards-strict external parsers carrying `SN`=serialNumber would be reparsed with different identity semantics on this path.

_Suggested:_ confirm no third-party strict-parser DNs flow through here; if they can, scope registry-first to platform-rendered DNs, or reserve `SN` for serialNumber.

<sub>flagged by Copilot CLI + superpowers</sub>

**Review by lubomirw — COMMENTED** [review]
Automated multi-reviewer triage — 5 findings (COMMENT). Sources: Copilot CLI, guided-pr-review, superpowers, hygiene. All findings anchored inline; nothing in overflow.

File: src/main/java/com/otilm/core/oid/OidHandler.java [hunk 3]
```diff
@@ -62,7 +93,10 @@ public static Map<String, String> getCodeToOidMap() {
     }
 
 
-    public static void removeCachedOid(OidCategory category, String oid) {
-        oidCache.get(category).remove(oid);
+    public static synchronized void removeCachedOid(OidCategory category, String oid) {
```
Comment (lubomirw) [inline src/main/java/com/otilm/core/oid/OidHandler.java:96]: **[Medium]** | Testing

The rewritten `removeCachedOid` (copy-on-write replace + `refreshRdnCodeLookup`) has no test verifying a code stops resolving after removal.

_Suggested:_ cache an RDN OID, assert its code resolves via `getOidForRdnCode`, call `removeCachedOid`, then assert it returns null and a DN using that code no longer parses.

<sub>flagged by superpowers</sub>

File: src/main/java/com/otilm/core/oid/OidHandler.java [hunk 4]
```diff
@@ -22,16 +25,44 @@ public static boolean isOid(String value) {
...
-    public static void cacheOidCategory(OidCategory category, Map<String, OidRecord> oidRecordMap) {
+    public static synchronized void cacheOidCategory(OidCategory category, Map<String, OidRecord> oidRecordMap) {
         oidCache.put(category, oidRecordMap);
+        refreshRdnCodeLookup(category);
+    }
+
+    public static synchronized void cacheOid(OidCategory category, String oid, OidRecord oidRecord) {
+        // Copy-on-write: published per-category maps are iterated lock-free by readers
+        // (getCodeToOidMap, style snapshots), so never mutate one in place.
+        Map<String, OidRecord> next = new HashMap<>(oidCache.get(category));
```
Comment (lubomirw) [inline src/main/java/com/otilm/core/oid/OidHandler.java:48]: **[Low]** | Error handling

`new HashMap<>(oidCache.get(category))` throws NPE when the category was never cached (also `removeCachedOid`, line 97). Pre-existing risk — old code did `oidCache.get(category).put/remove` with the same failure — so not a regression, but the refactor is a cheap place to guard.

_Suggested:_ `oidCache.getOrDefault(category, java.util.Map.of())`, or fail fast with `IllegalStateException` naming the uninitialized category.

<sub>flagged by Copilot CLI</sub>

File: src/main/java/com/otilm/core/oid/OidHandler.java [hunk 5]
```diff
@@ -22,16 +25,44 @@ public static boolean isOid(String value) {
...
+        next.put(oid, oidRecord);
+        oidCache.put(category, next);
+        refreshRdnCodeLookup(category);
     }
 
-    public static void cacheOid(OidCategory category, String oid, OidRecord oidRecord) {
-        oidCache.get(category).put(oid, oidRecord);
+    /** OID for an RDN code or alternative code, matched case-insensitively; {@code null} when unknown. */
+    public static String getOidForRdnCode(String code) {
```
Comment (lubomirw) [inline src/main/java/com/otilm/core/oid/OidHandler.java:55]: **[Nit]** | Testing

`getOidForRdnCode` is pure static logic but only exercised through the Spring-boot `PlatformX500NameStyleITest`; null-input and case-insensitive matching have no cheap unit test.

_Suggested:_ add an `OidHandler` unit test asserting `getOidForRdnCode(null)==null` and case-insensitive code/altCode resolution — no Spring context needed.

<sub>flagged by superpowers</sub>

**Comment by klaraf755** [comment]
The review comments were addressed in PR https://github.com/OmniTrustILM/core/pull/1868.
